frontend/cmd/cmd.go

```diff
@@ -0,0 +1,137 @@
+package cmd
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"os"
+	"os/signal"
+	"runtime/debug"
+	"syscall"
+
+	sdk "github.com/openshift-online/ocm-sdk-go"
+	"github.com/spf13/cobra"
+
+	"github.com/Azure/ARO-HCP/frontend/pkg/config"
+	"github.com/Azure/ARO-HCP/frontend/pkg/database"
+	"github.com/Azure/ARO-HCP/frontend/pkg/frontend"
+)
+
+type FrontendOpts struct {
+	clustersServiceURL string
+	insecure           bool
+
+	region string
+	port   int
+
+	useCache     bool
+	databaseName string
+	databaseURL  string
+}
+
+func NewRootCmd() *cobra.Command {
+	opts := &FrontendOpts{}
+	rootCmd := &cobra.Command{
+		Use:   "aro-hcp-frontend",
+		Args:  cobra.NoArgs,
+		Short: "Serve the ARO HCP Frontend",
+		Long: `Serve the ARO HCP Frontend
+
+	This command runs the ARO HCP Frontend. It communicates with Clusters Service and a CosmosDB
+
+	# Run ARO HCP Frontend locally to connect to a local Clusters Service at http://localhost:8000
+	./aro-hcp-frontend --database-name ${DB_NAME} --database-url ${DB_URL} --region ${REGION} \
+		--clusters-service-url "http://localhost:8000"
+`,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return opts.Run()
+		},
+	}
+
+	rootCmd.Flags().BoolVar(&opts.useCache, "use-cache", false, "leverage a local cache instead of reaching out to a database")
+	rootCmd.Flags().StringVar(&opts.databaseName, "database-name", os.Getenv("DB_NAME"), "database name")
+	rootCmd.Flags().StringVar(&opts.databaseURL, "database-url", os.Getenv("DB_URL"), "database url")
+	rootCmd.Flags().StringVar(&opts.region, "region", os.Getenv("REGION"), "Azure region")
+	rootCmd.Flags().IntVar(&opts.port, "port", 8443, "port to listen on")
+
+	rootCmd.Flags().StringVar(&opts.clustersServiceURL, "clusters-service-url", "https://api.openshift.com", "URL of the OCM API gateway.")
+	rootCmd.Flags().BoolVar(&opts.insecure, "insecure", false, "Skip validating TLS for clusters-service.")
+
+	rootCmd.MarkFlagsMutuallyExclusive("use-cache", "database-name")
+	rootCmd.MarkFlagsRequiredTogether("database-name", "database-url")
+
+	return rootCmd
+}
+
+func (opts *FrontendOpts) Run() error {
+	version := "unknown"
+	if info, ok := debug.ReadBuildInfo(); ok {
+		for _, setting := range info.Settings {
+			if setting.Key == "vcs.revision" {
+				version = setting.Value
+				break
+			}
+		}
+	}
+
+	logger := config.DefaultLogger()
+	logger.Info(fmt.Sprintf("%s (%s) started", frontend.ProgramName, version))
```

Comment:
slog's logger doesn't format strings like that, the first argument is the message, and following arguments are meant to be key/value pairs. I could do something like this for example if you prefer

```go
logger.Info("started", "program name", frontend.ProgramName, "version", version)
```

Reply:
Yes, sorry. I [meant](https://go.dev/play/p/-kY0k3Xva9j) to use it like you showed here.
 Otherwise, we lose the structural logging and we just have a plain string in the "msg" field, right?